src/psyclone/gocean1p0.py

```diff
@@ -1151,33 +1155,35 @@ def index_offset(self):
 
     def get_kernel_schedule(self):
         '''
-        Returns a PSyIR Schedule representing the GOcean kernel code.
-
-        :return: Schedule representing the kernel code.
+        :returns: a schedule representing the GOcean kernel code.
         :rtype: :py:class:`psyclone.gocean1p0.GOKernelSchedule`
-        '''
-        if self._kern_schedule is None:
-            astp = GOFparser2Reader()
-            self._kern_schedule = astp.generate_schedule(self.name, self.ast)
-            # TODO: Validate kernel with metadata (issue #288).
-        return self._kern_schedule
 
+        :raises InternalError: if the subroutine implementing this kernel
+            cannot be found in the associated Fortran parse tree.
 
-class GOFparser2Reader(Fparser2Reader):
-    '''
-    Sub-classes the Fparser2Reader with GOcean 1.0 specific
-    functionality.
-    '''
-    @staticmethod
-    def _create_schedule(name):
         '''
-        Create an empty KernelSchedule.
+        if self._kern_schedule:
+            return self._kern_schedule
 
-        :param str name: Name of the subroutine represented by the kernel.
-        :returns: New GOKernelSchedule empty object.
-        :rtype: py:class:`psyclone.gocean1p0.GOKernelSchedule`
-        '''
-        return GOKernelSchedule(name)
+        # Construct the PSyIR of the Fortran parse tree.
+        astp = Fparser2Reader()
+        psyir = astp.generate_psyir(self.ast)
+        # pylint: disable=import-outside-toplevel
+        from psyclone.domain.gocean.transformations import (
+            RaisePSyIR2GOceanKernTrans)
+        raise_trans = RaisePSyIR2GOceanKernTrans(self._metadata_name)
+        raise_trans.apply(psyir)
+        for routine in psyir.walk(Routine):
+            if routine.name == self.name:
+                break
+        else:
+            raise InternalError(
+                f"Failed to find kernel routine '{self.name}' in raised PSyIR."
+                f" This should have been caught by the validate method of "
+                f"RaisePSyIR2GOceanKernTrans.")
```

Comment:
I was a bit unsure about this (it's good to defend ourselves against future programming errors) but since this code is tightly coupled to the Raising transformation I've gone with your suggestion.